### ISSUE TITLE
How do I ungroup() without flattening?

### ISSUE THREAD

**Issue by Zythyr** [svgdotjs/svg.js#841]
I want to ungroup a node without flattening it. When I use the ungroup() method (http://svgjs.com/manipulating/#ungroup-flatten), it ends up flattening all the nodes inside the group, including the nested group. 

Below is the example of a nested group. I want to ungroup only the top level group (id=id="SvgjsG1008") such that after ungrouping, the children groups are still maintained. 

What I currently have:
```
<svg id="SvgjsSvg1001" width="100%" height="100%" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs">
    <defs id="SvgjsDefs1002"></defs>
    <g id="SvgjsG1008">
        <g id="SvgjsG1009">
            <rect id="SvgjsRect1010" width="50" height="50" x="0" y="0"></rect>
            <rect id="SvgjsRect1011" width="50" height="50" x="0" y="62.5"></rect>
        </g>
        <g id="SvgjsG1012">
            <rect id="SvgjsRect1013" width="50" height="50" x="62.5" y="0"></rect>
            <rect id="SvgjsRect1014" width="50" height="50" x="62.5" y="62.5"></rect>
        </g>
    </g>
</svg>
```

What I want: 
```
<svg id="SvgjsSvg1001" width="100%" height="100%" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs">
    <defs id="SvgjsDefs1002"></defs>
    <g id="SvgjsG1009">
        <rect id="SvgjsRect1010" width="50" height="50" x="0" y="0"></rect>
        <rect id="SvgjsRect1011" width="50" height="50" x="0" y="62.5"></rect>
    </g>
    <g id="SvgjsG1012">
        <rect id="SvgjsRect1013" width="50" height="50" x="62.5" y="0"></rect>
        <rect id="SvgjsRect1014" width="50" height="50" x="62.5" y="62.5"></rect>
    </g>
</svg>
```
Result of using the group.ungroup() function. It gives me same results even when I specify the depth. 
```
<svg id="SvgjsSvg1001" width="100%" height="100%" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs">
    <defs id="SvgjsDefs1002"></defs>
    <rect id="SvgjsRect1010" width="50" height="50" x="0" y="0" transform="matrix(1,0,0,1,0,0)"></rect>
    <rect id="SvgjsRect1011" width="50" height="50" x="0" y="62.5" transform="matrix(1,0,0,1,0,0)"></rect>
    <rect id="SvgjsRect1013" width="50" height="50" x="62.5" y="0" transform="matrix(1,0,0,1,0,0)"></rect>
    <rect id="SvgjsRect1014" width="50" height="50" x="62.5" y="62.5" transform="matrix(1,0,0,1,0,0)"></rect>
</svg>
```

Sample test code: https://codepen.io/anon/pen/jzxeZX 
```
var draw = SVG('drawing')

cols = 2
rows = 2
size = 50

bigGroup = draw.group()

for (i = 0; i < cols; i ++){
  group = draw.group()
  for (j = 0; j < rows; j ++){
    node = draw.rect(size,size).move(size*1.25*i,size*1.25*j)
    group.add(node)
    bigGroup.add(group)
  }
}

bigGroup.ungroup(bigGroup.parent())
//bigGroup.ungroup(bigGroup.parent(),5)
```

If I use the method below, then it lets me achieve what I want. However, wasn't ungroup() method intended to achieve the same as below? 
```
bigGroup.each(function(i, children) {
  this.putIn(bigGroup.parent())
})
bigGroup.remove()
```

**Comment by Fuzzyma** [comment]
Take a look at #827 where this feature was discussed. I will be im the 3.0 release. When you need it now you might want to cherry pick the commit for you.

**Comment by Zythyr** [comment]
@Fuzzyma Thanks. Glad to hear the feature is implemented in v3.0.0. 

This issue is resolved! :)
